### DISCUSSION TITLE
Possibility to change Drive monitoring

### DISCUSSION THREAD

**Discussion by jeremieLyvoc** [mathoudebine/turing-smart-screen-python#188] [Q&A]
Hi,

I'd like to know if it was possible to change the Hard Drive monitored ? 

I'm not very interested to monitor my system Drive.

Thanks !

**Comment by wsokc** [comment]
I'm looking into this too, apparently its only C drive,

**Comment by c0ldJS** [comment]
While you could manually alter the files provided by the project to do this, you'd then need to remember to remake the alteration each time you pull down a version update. And that's assuming you only want to keep eyes on a single storage device.

A theme I've been working on is reporting back on all storage devices by way of the `sensors_custom.py` file, where you can roll your own metric pulls. Here's an example snippet of what I've got in place currently:
![storage-example](https://github.com/user-attachments/assets/c2194fbd-7173-474e-9d54-abc4ed77021e)
It's not uniform for all devices just yet and work is at a standstill for now (see #551).

The wiki can get you [started](https://github.com/mathoudebine/turing-smart-screen-python/wiki/System-monitor-%3A-themes#add-custom-data-to-a-theme) on how to begin using `sensors_custom`. This method doesn't require or care about a specific `Hardware Monitoring` setting and you can use whatever tools local to the system that you enjoy using to ingest whatever it is you need. After crafting a class in `sensors_custom`, add its name in the dedicated `CUSTOM:` stanza of your theme's YAML with whatever options for drawing to screen (text output, line graphs, radials, etc.) that make sense.
